x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2287,6 +2288,33 @@ export class RulesClient {
     );
   }
 
+  public async clearExpiredSnoozes({ id }: { id: string }): Promise<void> {
+    const { attributes, version } = await this.unsecuredSavedObjectsClient.get<RawRule>(
+      'alert',
+      id
+    );
+
+    const snoozeSchedule = attributes.snoozeSchedule
+      ? attributes.snoozeSchedule.filter((s) => !isSnoozeExpired(s))
```

Comment:
It seems like `isSnoozeExpired` can throw, just curious to know what would happen if it does throw? I guess 1 invalid snooze could potentially prevent all other snoozes from being cleared. Maybe we should have some kind of recovery logic?